mkdocs/tests/nav_tests.py

```diff
@@ -610,3 +610,84 @@ def test_indented_toc_missing_child_title(self):
         self.assertEqual(str(site_navigation).strip(), expected)
         self.assertEqual(len(site_navigation.nav_items), 3)
         self.assertEqual(len(site_navigation.pages), 6)
+
+
+    def test_edit_url_basic(self):
+        """
+        Ensure that get_edit_url returns well formed URLs for simple base URLs
+        """
+
+        pages = [
+            'index.md',
+            'internal.md',
+            'sub/internal.md',
+            'sub1/sub2/internal.md',
+        ]
+
+        repo_url = 'http://example.com/'
+        edit_uri = 'edit/master/docs/'
+
+        site_navigation = nav.SiteNavigation(pages)
+
+        expected_results = (
+            repo_url + edit_uri + pages[0],
+            repo_url + edit_uri + pages[1],
+            repo_url + edit_uri + pages[2],
+            repo_url + edit_uri + pages[3],
+        )
+
+
+        for idx, page in enumerate(site_navigation.walk_pages()):
+            page.set_edit_url(repo_url, edit_uri)
+            self.assertEqual(page.edit_url, expected_results[idx])
+
+    def test_edit_url_query(self):
+        """
+        Ensure that get_edit_url returns well formed URLs for base URLs with
+        query strings
+        """
+
+        pages = [
+            'index.md',
+            'internal.md',
+            'sub/internal.md',
+            'sub1/sub2/internal.md',
+        ]
+
+        # Test repo_url ending without '/'
+
+        repo_url = 'http://example.com/workspace'
+        edit_uri = '?path=edit/master/docs/'
+
+        site_navigation = nav.SiteNavigation(pages)
+
+        expected_results = (
+            repo_url + edit_uri + pages[0],
```

Comment:
This would equal `http://example.com/workspace?path=edit/master/docs/index.md`. Is that correct or do we need to add an extra slash in there: `repo_url + '/' + edit_uri + pages[0]`. I'm not sure which way you intended. See my other comment below.